src/librustc/diagnostics.rs

```diff
@@ -772,6 +776,24 @@ struct Foo<'a> {
     x: &'a str,
 }
 ```
+
+Implementations need their own lifetime declarations:
+    
+```
+// error, undeclared lifetime
+impl Foo<'a> {
+    ...
+}
+```
+
+Which are declared like this:
+
+```
+// correct
```

Comment:
These code samples are tested, so please include the struct definition in this sample to make sure it properly compiles.